### PR TITLE
Add order note at the end of both invoices following current UI patterns.

### LINKED ISSUE #9179: Show notes written by admins on invoices
## Description
**- As a:** enterprise user
**- On page:** On an invoice (PDF)
**- I want to be able to do:** See the note that I have added thanks to 
https://github.com/openfoodfoundation/openfoodnetwork/issues/9178

The note should appear at the end of the rest of the invoice content, with the same UX than the payment description for example : 

Title + note below. The Title should be "Additional information".

## Acceptance Criteria & Tests

1. Log in as an enterprise user
2.  Add a note on an order
3.  Check it appears when you generate the invoice
4. Delete the note + check orders without notes : notion should appear on invoices

Repeat with super admin

:warning: be sure to apply it on BOTH invoices template.

### PR DESCRIPTION
**Pull request by apricot12**
Add translation for note title in invoice

Update invoice spec to also test for order note

#### What? Why?

Closes #9179 

It shows the newly created [order note](https://github.com/openfoodfoundation/openfoodnetwork/issues/9178) under both the present invoice templated following current UI patterns.

image

#### What should we test?

- Visit `admin/orders` page as admin/superadmin.
- Edit any one of the available orders to have a note.
- Click on the `Actions` button and select `Print invoice`.
- Ensure that the newly added note is visible at the very bottom of the invoice like shown in the screenshot above.
- Change to the alternate invoice template and ensure the note is still visible.
- Remove the note through the orders edit page.
- Again, print invoice under the `Actions` button.
- See that the note is no longer visible along with the title for both templates.

#### Release notes

Changelog Category: User facing changes

### CODE REVIEW

**Comment by binarygit** [comment]
Hello @jibees is on holiday so I thought I'd start the first review round

File: app/views/spree/shared/_order_note.html.haml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+%p.callout{style: "margin-top: 30px"}
```
Comment (binarygit) [inline app/views/spree/shared/_order_note.html.haml:1]: Why inline styles? I think it's better to have these in the stylesheet itself.

Comment (apricot12) [reply app/views/spree/shared/_order_note.html.haml:1]: Oh I agree that its generally better to add these in the stylesheet but I was just following the pattern here. 🤣

File: app/services/permitted_attributes/checkout.rb [hunk 2]
```diff
@@ -12,9 +12,9 @@ def call
           :email, :special_instructions,
           :existing_card_id, :shipping_method_id,
           { payments_attributes: [
-            :payment_method_id,
-            { source_attributes: PermittedAttributes::PaymentSource.attributes }
-          ],
+              :payment_method_id,
+              { source_attributes: PermittedAttributes::PaymentSource.attributes }
+            ],
```
Comment (mkllnk) [inline app/services/permitted_attributes/checkout.rb:17]: It would be better to keep these formatting changes in a separate commit.

File: app/views/spree/shared/_order_note.html.haml (hunk 1 above)
Comment (mkllnk) [reply app/views/spree/shared/_order_note.html.haml:1]: Yes, I think that you are both right. The special case here is that this HTML is used to generate a PDF and the styles are not re-used anywhere. So it's okay. I don't actually know if separate CSS works but it would be nice.

**Review by mkllnk — APPROVED** [review]
Thank you!

File: app/views/spree/shared/_order_note.html.haml (hunk 1 above)
Comment (binarygit) [reply app/views/spree/shared/_order_note.html.haml:1]: Ah I see. It's okay then I think :smile:

**Comment by filipefurtad0** [comment]
Hi @apricot12, 

We can see that this PR displays the text box content in the invoices: 

image

For 280 characters, this is how it looks like:

image

Looks good, thanks!
Merging.
